### PR TITLE
feat(TBD-5453): Add CDH 5.13 distribution using new mechanism (DI & M/R)

### PR DESCRIPTION
**Pull request by zafkirTalend**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

No CDH 5.13 distribution using new mechanism (DI & M/R)

**What is the new behavior?**

Add support for CDH 5.13 distribution using new mechanism (DI & M/R)

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_java.xml [hunk 1]
```diff
@@ -878,6 +878,19 @@
...
+				UrlPath="platform:/plugin/org.talend.libraries.pig/lib/elephant-bird-pig-4.9.jar"
+				REQUIRED_IF="(STORE=='SEQUENCEFILESTORAGE')" />
+			<IMPORT NAME="elephant-bird-core-4.9.jar" MODULE="elephant-bird-core-4.9.jar"
+				MVN="mvn:org.talend.libraries/elephant-bird-core-4.9/6.0.0"
+				UrlPath="platform:/plugin/org.talend.libraries.pig/lib/elephant-bird-core-4.9.jar"
+				REQUIRED_IF="(STORE=='SEQUENCEFILESTORAGE')" />
+			<IMPORT NAME="elephant-bird-hadoop-compat-4.9.jar" MODULE="elephant-bird-hadoop-compat-4.9.jar"
+				MVN="mvn:org.talend.libraries/elephant-bird-hadoop-compat-4.9/6.0.0"
+				UrlPath="platform:/plugin/org.talend.libraries.pig/lib/elephant-bird-hadoop-compat-4.9.jar"
+				REQUIRED_IF="(STORE=='SEQUENCEFILESTORAGE')" />
```
Comment (amarouni) [inline main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_java.xml]: Why we added these jars here ? Can you add a small comment explaining what we did and why ?

File: main/plugins/org.talend.hadoop.distribution/META-INF/eclipse.inf [hunk 2]
```diff
@@ -0,0 +1 @@
+jarprocessor.exclude.children=true
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/META-INF/eclipse.inf]: why is this file included ??

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+        // normally bestTemplateBean can't be null here
+        DynamicTemplateAdapter templateAdapter = new DynamicTemplateAdapter(bestTemplateBean, configuration);
+        templateAdapter.adapt(monitor);
+        IDynamicPlugin dynamicPlugin = templateAdapter.getDynamicPlugin();
+
+        return dynamicPlugin;
+    }
+
+    @Override
+    public void regist(IDynamicPlugin dynamicPlugin, IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java]: is this supposed to say `register` ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+            }
+        }
+
+        templateBeansCache = templates;
+
+        return templateBeansCache;
+    }
+
+    @Override
+    public List<IDynamicPlugin> getAllBuildinDynamicPlugins(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java]: `Buildin` to `Builtin`

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+            ServiceRegistration osgiService = DynamicServiceUtil.registOSGiService(context,
+                    distributionTemplate.getServices().toArray(new String[0]), distributionTemplate, null);
+            registedOsgiServiceMap.put(id, osgiService);
+        } finally {
+            plugin.setPluginConfiguration(pConfiguration);
+        }
+    }
+
+    @Override
+    public void registAllBuildin(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+            try {
+                regist(dynamicPlugin, monitor);
+            } catch (Throwable e) {
+                ExceptionHandler.process(e);
+            }
+        }
+    }
+
+    @Override
+    public void unregistAllBuildin(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+                ExceptionHandler.process(e);
+            }
+        }
+    }
+
+    abstract protected IDynamicDistributionTemplate initTemplate(DynamicPluginAdapter pluginAdapter, IDynamicMonitor monitor)
+            throws Exception;
+
+    @Override
+    public void unregist(IDynamicPlugin dynamicPlugin, IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistribution.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+                }
+            }
+        }
+
+        // normally bestDistribution can't be null here
+        return bestDistribution.buildDynamicPlugin(monitor, configuration);
+    }
+
+    @Override
+    public List<IDynamicPlugin> getAllBuildinDynamicPlugins(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java]: buildin to builtin

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+                    }
+                } catch (Exception e) {
+                    ExceptionHandler.process(e);
+                }
+            }
+        }
+        return dynamicPlugins;
+    }
+
+    protected List<IDynamicDistribution> getAllRegistedDynamicDistributions(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+                IDynamicDistribution service = bc.getService(sr);
+                registedDynamicDistributions.add(service);
+            }
+        }
+
+        return registedDynamicDistributions;
+    }
+
+    @Override
+    public void registAllBuildin(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+            try {
+                dynamicDistribution.registAllBuildin(monitor);
+            } catch (Throwable e) {
+                ExceptionHandler.process(e);
+            }
+        }
+    }
+
+    @Override
+    public void unregistAllBuildin(IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+                        ExceptionHandler.process(e);
+                    }
+                }
+            }
+        }
+        return templateIdMap.get(templateId);
+    }
+
+    @Override
+    public void regist(IDynamicPlugin dynamicPlugin, IDynamicMonitor monitor) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/AbstractDynamicDistributionsGroup.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/DynamicDistributionManager.java [hunk 13]
```diff
@@ -0,0 +1,596 @@
...
+                }
+            } else {
+                fileList.add(file.getCanonicalPath());
+            }
+        }
+
+        return fileList;
+    }
+
+    public void registAll(IDynamicMonitor monitor, boolean resetCache) throws Exception {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/DynamicDistributionManager.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/IDynamicDistribution.java [hunk 14]
```diff
@@ -12,14 +12,40 @@
...
+
+    public List<TemplateBean> getTemplates(IDynamicMonitor monitor) throws Exception;
+
+    public IDynamicPlugin buildDynamicPlugin(IDynamicMonitor monitor, DynamicConfiguration configuration) throws Exception;
+
+    public List<IDynamicPlugin> getAllBuildinDynamicPlugins(IDynamicMonitor monitor) throws Exception;
+
+    public String getDistributionName();
+
+    public void regist(IDynamicPlugin dynamicPlugin, IDynamicMonitor monitor) throws Exception;
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/IDynamicDistribution.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/IDynamicDistributionsGroup.java [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+    public List<TemplateBean> getAllTemplates(IDynamicMonitor monitor) throws Exception;
+
+    public IDynamicPlugin buildDynamicPlugin(IDynamicMonitor monitor, DynamicConfiguration configuration) throws Exception;
+
+    public List<IDynamicPlugin> getAllBuildinDynamicPlugins(IDynamicMonitor monitor) throws Exception;
+
+    public List<String> getAllVersions(IDynamicMonitor monitor) throws Exception;
+
+    public void registAllBuildin(IDynamicMonitor monitor) throws Exception;
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/IDynamicDistributionsGroup.java]: is this supposed to say register ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/adapter/DynamicDistriConfigAdapter.java [hunk 16]
```diff
@@ -17,10 +17,14 @@
...
 import org.talend.hadoop.distribution.dynamic.bean.TemplateBean;
+import org.talend.hadoop.distribution.dynamic.util.DynamicDistributionUtils;
+import org.talend.hadoop.distribution.i18n.Messages;
+import org.talend.repository.ProjectManager;
 
 /**
- * DOC cmeng  class global comment. Detailled comment
+ * DOC cmeng class global comment. Detailled comment
  */
 public class DynamicDistriConfigAdapter extends AbstractDynamicAdapter {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/adapter/DynamicDistriConfigAdapter.java]: is this supposed to say Distrib ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/spark/SparkClassPathUtils.java [hunk 17]
```diff
@@ -47,5 +47,27 @@ public static String generateSparkJarsPaths(List<String> commandLineJarsPaths, S
...
+     * module group name. Then it proceeds to extract all jars' names from the module group. The list of all Spark jars'
+     * names is then used to generate a new list, from the list of all classpath jars, that only contains classpath jars
+     * that are present in the Spark module group. The final list is then transformed to java like classpath string
+     * 
+     * @return A string with all of the Spark jars.
+     * @param commandLineJarsPaths the list of all the job's classpath jars
+     * @param sparkModuleGroupName The Spark module group name
+     * @return the string representation of the classpath with Spark jars paths
+     */
+    public static String generateSparkJarsPathsWithNames(List<String> commandLineJarsPaths, String sparkModuleGroupName) {
```
Comment (amarouni) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/spark/SparkClassPathUtils.java]: Is the javadoc above is still relevant ? was it updated for this method ?

Comment (yu-talend) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/spark/SparkClassPathUtils.java]: Yes, we slightly changed to do with jar "name" from jar "id", so we have also modified the doc.

File: main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_java.xml (hunk 1 above)
Comment (yu-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_java.xml]: Zak will take a look at this later because Pig still doesn't work for me.

File: main/plugins/org.talend.hadoop.distribution/META-INF/eclipse.inf (hunk 2 above)
Comment (yu-talend) [reply main/plugins/org.talend.hadoop.distribution/META-INF/eclipse.inf]: @cmeng-talend you fixed it in https://github.com/Talend/tbd-studio-se/commit/c14614cea9822055b241c44750311f34b597795e ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/DynamicDistributionManager.java (hunk 13 above)
Comment (yu-talend) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/dynamic/DynamicDistributionManager.java]: This class's interface is defined in tcommon-studio-se, so I also changed here https://github.com/Talend/tcommon-studio-se/commit/5e971018ff447689e4f6667e7eaaeac5c0ec106f

File: main/plugins/org.talend.hadoop.distribution/META-INF/eclipse.inf (hunk 2 above)
Comment (cmeng-talend) [reply main/plugins/org.talend.hadoop.distribution/META-INF/eclipse.inf]: yes, just keep it same like other plugins
